### ISSUE TITLE
Split flang.exe into flang-driver

### ISSUE THREAD

**Issue by xoviat** [conda-forge/clangdev-feedstock#32]
From https://github.com/conda-forge/numpy-feedstock/issues/79#issuecomment-366313872

> One idea is to create a new flang-driver package that is just the flang executable and have it depend on clangdev (similar to how we split libflang out).

cc @isuruf

**Comment by isuruf** [comment]
Does `flang.exe` depend on other libraries in `clangdev` package?

**Comment by xoviat** [comment]
Yes. So the run requirements for `flang-driver` would be `clangdev` and `flang`. The run requirements for `flang` would be `flang-driver`.

**Comment by h-vetinari** [comment]
Not sure where to put this best, but since I see this issue: we should probably create a new feedstock for flang starting with LLVM 14 (code generation will not be complete, but should help shake out issues)?
